kong/db/dao/init.lua

```diff
@@ -145,6 +146,24 @@ local function validate_options_value(options, schema, context)
     errors.ttl = fmt("cannot be used with '%s'", schema.name)
   end
 
+  if schema.tags == true and options.tags ~= nil then
+    if context ~= "select" then
```

Comment:
does the `options.tags` filter also work in generated methods for relationships, such as `db.targets:each_for_upstream({ id = U.id }, nil, { tags = T })`? (i.e. "give me all targets of upstream X that have tag T") — there needs to be a test in the suite answering this question :)

Reply:
each_for_* will not support tags because it will not parse "?tags=" from Admin API, and uses a different pager `page_for_*` other than the tag-enabled function `page` used by normal pagination. I'll add a unit test on each_for_upstream to ensure the functionality doesn't exist.